Bmt2 45 separate authorization and authentication logic to auth service

hovnaz opened the pull request.

hovnaz committed:
implementation crud functional auth service and user service

hovnaz committed:
change some types entity id Long to long and User response Date to LocaleDate

Mushegh128 commented on black-mulberry-core/src/main/java/com/black/mulberry/core/exception/UserEmailConflict.java:
```diff
@@ -0,0 +1,23 @@
+public class UserEmailConflict extends RuntimeException {
+
+    public UserEmailConflict() {
+    }
```
I don't recommend using constructors that don't take arguments as message. lets delete them.

Mushegh128 commented on black-mulberry-core/src/main/java/com/black/mulberry/core/mapper/UserRegistrationMapper.java:
```diff
@@ -2,21 +2,21 @@
 @Service
 @RequiredArgsConstructor
-public class UserSignUpMapper implements BaseMapper<User, UserSignUpRequest, UserResponse> {
+public class UserRegistrationMapper implements BaseMapper<User, UserRegistrationRequest, UserResponse> {
```
i think registration reguest DATA will be different from simple user reguest DATA . Reconsider the creation of a simple user request and each object of use for its intended purpose.
you can use overloading too !!!

Mushegh128 commented on black-mulberry-core/src/main/java/com/black/mulberry/core/util/UserServiceUtil.java:
```diff
@@ -0,0 +1,21 @@
+@Component
+@RequiredArgsConstructor
+public class UserServiceUtil {
+    private final UserRepository userRepository;
```
why its util? in general ist ok, buts it can be better - lets rename UserSServiceUtil to UserServiceSupport in package ......service.support.UserServiceSupport.java and we can change @Component anotation to @Service.

Mushegh128 commented on black-mulberry-data-transfer/src/main/java/com/black/mulberry/data/transfer/response/UserAuthResponse.java:
```diff
@@ -0,0 +1,14 @@
+@AllArgsConstructor
+@Data
+public class UserAuthResponse {
+    private UserResponse user;
```
why we created UserAuthResponse when its include the same data that includes in UserResponse? please check this

hovnaz replied:
We left this to add the jwt token in the next task, that's why I returned it like this․
If you think i did something wrong please share

hovnaz committed:
1.rename UserServiceUtil to UserServiceSupport
2. remove empty argument method from Exceptions
3. delete EmailExistException this is no used

hovnaz committed:
1. add log for UserServiceImpl and AuthServiceImpl
2. and some optimize code

hovnaz committed:
AuthServiceImpl in registration method added return type UserRegistrationResponse
